images/terraform/repo_pair/main.tf

```diff
@@ -0,0 +1,14 @@
+resource "aws_ecr_repository" "private" {
+  name = "${var.namespace}/${var.repo_name}"
+}
+
+resource "aws_ecrpublic_repository" "public" {
+  provider = aws.ecr_public
```

Comment:
In case you're wondering: ECR Public is another global resource, like IAM or CloudFront, and it lives in us-east-1.